pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -328,7 +328,20 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 	}
 
 	if !infraObj.data.Bool("status", "jobComplete") && infraObj.data.String("status", "failureReason") == "" {
-		return obj, fmt.Errorf("cannot delete machine %s because create job has not finished", infraObj.meta.GetName())
+		job, err := h.getJobFromInfraMachine(infraObj)
+		if job != nil || (err != nil && !apierrors.IsNotFound(err)) {
+			logrus.Debugf("[MachineProvision] create job for %s not finished, job was found and the error was not nil and was not an isnotfound", key)
+			logrus.Tracef("[MachineProvision] create job object for %s was %+v", key, job)
+			if job != nil {
+				// enqueue the job to force-reconcile the condition
+				h.jobController.Enqueue(job.Namespace, job.Name)
+			}
+			return obj, fmt.Errorf("cannot delete machine %s because create job has not finished", infraObj.meta.GetName())
+		}
+		// If the job is not found, go ahead and proceed with machine deletion
+		if apierrors.IsNotFound(err) {
+			return obj, nil
+		}
```

Comment:
```suggestion
		if apierrors.IsNotFound(err)) {
			// If the job is not found, go ahead and proceed with machine deletion
			return obj, nil
		} else if err != nil {
			return obj, err
		}
		logrus.Debugf("[MachineProvision] create job for %s not finished, job was found and the error was not nil and was not an isnotfound", key)
		logrus.Tracef("[MachineProvision] create job object for %s was %+v", key, job)

		// enqueue the job to force-reconcile the condition
		h.jobController.Enqueue(job.Namespace, job.Name)
		}
		
		logrus.Infof("[MachineProvision] cannot delete machine %s because job %s is not finished", infraObj.meta.GetName(), job.Name)
		return obj, generic.ErrSkip)
		}
```